content/articles/language-integrated-query-in-c#/index.md.md

```diff
@@ -0,0 +1,156 @@
+            prevX1 = prevX2;
+            prevX2 = temp;
+            yield return temp;
+        }
+    }
+}
+```
+Eager evaluation is the evaluation method used by most conventional programming languages.
+```c#
+IEnumerable<int> myMethod()
+{
+    for(int i=0; i <=900; i++)
+    {
+        yield return I;
+    }
+}
+...
+
+var ex = myMethod.Take(2);
+var list = ex.ToList();
+```
+The yield keyword allows the collection to be evaluated gradually. Eager evaluation is the process of forcing the entire collection into memory. When you call ToList, ToDictionary, or ToArray, the enumeration is evaluated immediately, and all of the elements are returned in a collection.
+
+### Identifying places in your code where LINQ queries may have performance difficulties
+The sample that we have created above performs an out shuffle, with the top and bottom cards remaining unchanged on every run. Let's make one alteration. instead of an out shuffle, we'll employ an in shuffle, in which all 52 cards shift positions. As a result, the top half's final card becomes the bottom half's final card. This is a straightforward modification to a single line of code. Switch the Take and Skip locations in the current shuffle query to update it. The top and bottom half of the deck will now be in reverse order.
+```c#
+shuffle = shuffle.Skip(26).InterleaveSequenceWith(shuffle.Take(26));
+```
+When you execute the program once more, you will note that the deck 'takes 52 iterations to reshuffle itself. This is contributed to by several factors. You may address one of the key causes of this reduction in performance which is the inefficient lazy evaluation. Note that we used a LINQ query to create the original deck. The three queries of LINQ on the previous deck are used to generate each shuffle. All of this occurs at a glacial speed. This indicates that the sequence is repeated each time it is requested. By the moment you reach the fifty second iteration, you will have regenerated the initial deck severaly i.e more than once. Let's make a log to show how this works. Then you'll take care of it.
+
+Type or copy the method below into your Extensions.cs file. This extension method results into the creation of a new file which is named as debug.log in your project directory and logs the query that is presently being run. This extension method can be used to indicate the completion of a query.
+```c#
+public static IEnumerable<T> LogQuery<T>
+    (this IEnumerable<T> sequence, string x)
+{
+   
+    using (var program = File.AppendText("debug.log"))
+    {
+        program.WriteLine($"Executing Query {x}");
+    }
+
+    return sequence;
+}
+```
+In the above code, do not forget to add the input-output extension so that the program will execute.
+```c#
+using System.IO;
+```
+### Conclusion
+In conclusion, we have got to see some working functionalities of the LINQ in C# and how they are implemented in say a program or solving some real-life problems.

```

Comment:
Change to "In conclusion, we have got to see some working functionalities of the LINQ in C# and how to implement them in a program or solve real-life problems."